search-app/src/cmr/search/services/parameters/converters/geometry.clj

```diff
@@ -35,16 +38,30 @@
   [^LineString line-string]
   (rr/ords->ring :geodetic (coords->ords (.getCoordinates line-string))))
 
+(defn force-ccw-orientation
+  "Forces a LineString to be in counter-clockwise orientation"
+  [^LineString line-string winding]
+  (let [coords (.getCoordinates line-string)]
+    (if (and 
+          (> (count coords) 3)
+          (= winding :cw))
+      (.reverse line-string)
+      line-string)))
+
 (defn polygon->shape
-  "Convert a JTS Polygon to a spatial lib shape that can be used in a Spatial query"
-  [^Polygon polygon]
-  (let [boundary-ring (.reverse (.getExteriorRing polygon))
+  "Convert a JTS Polygon to a spatial lib shape that can be used in a Spatial query.
+  The `context` map can be used to provide information about winding. Accepted keys
+  are `:boundary-winding` and `:hole-winding`. Accepted values are `:cw` and `ccw`."
+  [^Polygon polygon context]
+  (let [boundary-ring (force-ccw-orientation (.getExteriorRing polygon) (:boundary-winding context))
         num-interior-rings (.getNumInteriorRing polygon)
+        _ (debug (format "NUM INTERIOR RINGS: [%d]" num-interior-rings))
```

Comment:
Nit style: A `let` block has an implicit do. You can move both of the debug log statements out of the bindings.

```
(let [foo bar]
  (debug "debug 1")
  (debug "debug 2")
  (poly/polygon :geodetic (map line-string-ring->ring all-rings))))
```

Reply:
Done

Reply:
Done